src/game/Spell.cpp

```diff
@@ -4193,6 +4231,9 @@ SpellCastResult Spell::CheckCast(bool strict)
             return SPELL_FAILED_NOT_READY;
     }
 
+    if (!m_caster->isAlive() && m_caster->GetTypeId() == TYPEID_PLAYER && !m_spellInfo->HasAttribute(SPELL_ATTR_CASTABLE_WHILE_DEAD))
```

Comment:
also to call one over the other horrible to read is a bit pver the top, I would argue that my design choise is much easier to read and wrap your head around logically, but hey, we all think differently.